src/tree/tests.rs

```diff
@@ -7,21 +7,28 @@
 use crate::tree::{
     PoseidonAnnotation, PoseidonLeaf, PoseidonMaxAnnotation, PoseidonTree,
 };
-use canonical::Canon;
 use canonical_derive::Canon;
-use canonical_host::MemStore;
 use core::borrow::Borrow;
 use dusk_bls12_381::BlsScalar;
 use dusk_hades::{ScalarStrategy, Strategy};
-use rand::{CryptoRng, RngCore};
+use microkelvin::Keyed;
+use rand_core::{CryptoRng, RngCore};
 
-#[derive(Debug, Default, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Canon)]
+#[derive(
+    Debug, Default, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Canon,
+)]
 pub struct MockLeaf {
     s: BlsScalar,
     pub pos: u64,
     pub expiration: u64,
 }
 
+impl Keyed<u64> for MockLeaf {
+    fn key(&self) -> &u64 {
+        &self.pos
+    }
+}
+
 impl MockLeaf {
     pub fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
```

Comment:
If it's never used as the CI is saying, and never exported since it's a test, should be removed? Or it's used in the later PRs?

Reply:
It's used later and the CI doesn't give anymore warnings.